MapboxNavigation/NavigationMapView.swift

```diff
@@ -340,16 +340,20 @@ open class NavigationMapView: MGLMapView, UIGestureRecognizerDelegate {
             return
         }
         
+        let centerUserCourseView = { [weak self] in
+            guard let point = self?.convert(location.coordinate, toPointTo: self) else { return }
+            self?.userCourseView.center = point
+        }
+        
         if tracksUserCourse {
+            centerUserCourseView()
```

Comment:
This is the smoothest implementation I found so far, to make user puck slide in place gracefully...

Reply:
#2211 removed the line that set the view’s `center`. Can you check that it doesn’t introduce any regressions related to that PR? Thanks!

Reply:
Issue description and reproduction steps for #2211 are a bit cryptic for me, but as I understood, the issue was about incorrect puck positioning during navigation, together with zooming/panning/rotating map view. I checked that user puck is located right on the route regardless of camera position and switching to/from course tracking mode. Is that what #2211 was about?

Reply:
Unfortunately, I’m not super clear on it myself; I’m not entirely convinced that #2211 was the right fix for #2145, and it definitely wasn’t a fix for mapbox/mapbox-gl-native#15232 or mapbox/mapbox-gl-native#15233.

The main things to watch out for are https://github.com/mapbox/mapbox-navigation-ios/issues/2145#issuecomment-512611403 and https://github.com/mapbox/mapbox-navigation-ios/issues/2145#issuecomment-535475988.

Reply:
I could not reproduce symptoms described in linked comments...

Reply:
If I understand correctly, the Nav SDK is not using the Maps SDK user location annotation, which means it will not account for `contentInset` changes. I think this pr is trying to update the user location annotation considering the "insets".